docs/examples/Electromagnetic Phenomena/emwave_animation.py

```diff
@@ -0,0 +1,169 @@
+# dt: value by which t is incremented for each call of timer_callback
+# w: angular frequency (default = 0.1)
+# d: phase angle (default = 0.002)
+
+
+radius_waves = 0.02
+npoints = 800
+wavelength = 2
+k = np.pi*2/wavelength
+t = 0
+dt = 0.1
+w = 0.1
+d = 0.002
+
+###############################################################################
+# Creating a scene object and configuring the camera's position
+scene = window.Scene()
+scene.set_camera(position=(6, 5, 10), focal_point=(0.0, 0.0, 0.0),
+                 view_up=(0.0, 0.0, 0.0))
+showm = window.ShowManager(scene,
+                           size=(800, 600), reset_camera=True,
+                           order_transparent=True)
+showm.initialize()
+
+###############################################################################
+# Creating a yellow colored arrow to show the direction of propagation of
+# electromagnetic wave
+centers = np.array([[-3, 0, 0]])
+directions = np.array([[1, 0, 0]])
+heights = np.array([6.4])
+arrow_actor = actor.arrow(centers, directions, window.colors.yellow, heights,
+                          resolution=20, tip_length=0.06, tip_radius=0.012,
+                          shaft_radius=0.005)
+scene.add(arrow_actor)
+
+
+###############################################################################
+# Creating point actor that renders the magnetic field
+x = np.linspace(-3, 3, npoints)
+y = np.sin(k*x - w*t + d)
+z = np.array([0 for i in range(npoints)])
+
+pts = np.array([(a, b, c) for (a, b, c) in zip(x, y, z)])
+colors = window.colors.red
+wave_actor1 = actor.point(pts, colors, point_radius=radius_waves)
+scene.add(wave_actor1)
+
+vertices = utils.vertices_from_actor(wave_actor1)
+vcolors = utils.colors_from_actor(wave_actor1, 'colors')
+no_vertices_per_point = len(vertices)/npoints
+initial_vertices = vertices.copy() - \
+    np.repeat(pts, no_vertices_per_point, axis=0)
+
+
+###############################################################################
+# Creating point actor that renders the electric field
+xx = np.linspace(-3, 3, npoints)
+yy = np.array([0 for i in range(npoints)])
+zz = np.sin(k*xx - w*t + d)
+
+pts2 = np.array([(a, b, c) for (a, b, c) in zip(xx, yy, zz)])
+colors2 = window.colors.blue
+wave_actor2 = actor.point(pts2, colors2, point_radius=radius_waves)
+scene.add(wave_actor2)
+
+vertices2 = utils.vertices_from_actor(wave_actor2)
+vcolors2 = utils.colors_from_actor(wave_actor2, 'colors')
+no_vertices_per_point2 = len(vertices2)/npoints
+initial_vertices2 = vertices2.copy() - \
+    np.repeat(pts2, no_vertices_per_point2, axis=0)
+
+
+###############################################################################
+# Initializing text box to display the title of the animation
+tb = ui.TextBlock2D(bold=True, position=(160, 90))
+tb.message = "Electromagnetic Wave"
+scene.add(tb)
+
+###############################################################################
+# end is used to decide when to end the animation
+end = 1000
```

Comment:
For doc generation, can you make it shorter. thanks!

Reply:
I suppose you meant that I reduce the value of end? I reduced end from 1000 to 300.

Reply:
Yes, and thank you